R/utilities-writing-report.R

```diff
@@ -1,393 +1,400 @@
+
+  logWorkflow(
+    message = paste0("Report '", fileName, "' was initialized successfully."),
+    pathFolder = logFolder
+  )
+  return(invisible())
+}
+
+#' @title addFigureChunk
+#' @description Add a figure in a .md document
+#' @param fileName name of .md file
+#' @param figureFileRelativePath path to figure relative to working directory
+#' @param figureFileRootDirectory working directory
+#' @param figureCaption caption of figure
+#' @param logFolder folder where the logs are saved
+#' @export
+addFigureChunk <- function(fileName,
+                           figureFileRelativePath,
+                           figureFileRootDirectory,
+                           figureCaption = "",
+                           logFolder = getwd()) {
+  # For a figure path to be valid in markdown using ![](#figurePath)
+  # %20 needs to replace spaces in that figure path
+  mdFigureFile <- gsub(pattern = "[[:space:]*]", replacement = "%20", x = figureFileRelativePath)
+  mdText <- c(
+    "",
+    paste0("![", figureCaption, "](", mdFigureFile, ")"),
+    ""
+  )
+
+  fileObject <- file(fileName, encoding = "UTF-8", open = "at")
+  write(mdText, file = fileObject, append = TRUE, sep = "\n")
+  close(fileObject)
+
+  usedFilesFileName <- sub(pattern = ".md", replacement = "-usedFiles.txt", fileName)
+  fileObject <- file(usedFilesFileName, open = "at", encoding = "UTF-8")
+  write(file.path(figureFileRootDirectory, figureFileRelativePath), file = fileObject, append = TRUE, sep = "\n")
+  close(fileObject)
+
+  logWorkflow(
+    message = paste0("Figure path '", figureFileRelativePath, "' added to report '", fileName, "'."),
+    pathFolder = logFolder,
+    logTypes = LogTypes$Debug
+  )
+  return(invisible())
+}
+
+#' @title addTableChunk
+#' @description Add a table in a .md document
+#' @param fileName name of .md file
+#' @param tableFileRelativePath path to table relative to working directory
+#' @param tableFileRootDirectory working directory
+#' @param digits number of decimal digits in displayed numbers
+#' @param scientific logical defining if displayed numbers use scientific writing
+#' @param logFolder folder where the logs are saved
+#' @export
+addTableChunk <- function(fileName,
+                          tableFileRelativePath,
+                          tableFileRootDirectory,
+                          digits = NULL,
+                          scientific = NULL,
+                          logFolder = getwd()) {
+  # The function `formatNumerics` is now used by addTableChunk
+  # colClasses = "character" is not needed anymore to enforce all table elements to be 'character'
+  table <- read.csv(
+    file.path(tableFileRootDirectory, tableFileRelativePath),
+    check.names = FALSE,
+    #colClasses = "character",
+    fileEncoding = "UTF-8",
+    stringsAsFactors = FALSE
+  )
+  table <- formatNumerics(
+    table,
+    digits = digits %||% reEnv$formatNumericsDigits,
+    scientific = scientific %||% reEnv$formatNumericsScientific
+    )
```

Comment:
@msevestre `formatNumerics` is now applied only when added to the .md report.
The results exported in .csv have all the necessary digits.
Also, let me know if I should expose the `formatNumerics` method (which is now different from what is implemented in `ospsuite` package with no need for `nsmall`)